workers/tasks/search.js

```diff
@@ -53,31 +53,18 @@ export const setPubSearchData = (pubId) => {
 };
 
 export const updateCommunityData = (communityId) => {
-	const getObjectIds = (index) => {
-		return new Promise((resolve, reject) => {
-			const browser = index.browseAll('', {
-				filters: `communityId:${communityId}`,
-				attributesToRetrieve: ['objectId'],
-			});
-			let objectIds = [];
-
-			browser.on('result', (content) => {
-				objectIds = objectIds.concat(
-					content.hits.map((hit) => {
-						return hit.objectID;
-					}),
-				);
-			});
-
-			browser.on('end', () => {
-				resolve(objectIds);
-			});
-
-			browser.on('error', (err) => {
-				reject(err);
-			});
+	const getObjectIds = async (index) => {
+		let hits = [];
+		await index.browseObjects({
+			filters: `communityId:${communityId}`,
+			attributesToRetrieve: ['objectId'],
+			batch: (batch) => {
+				hits = hits.concat(batch);
+			},
```

Comment:
Algolia uses an interesting mix of streaming and Promise APIs here.

Reply:
If I had to guess it's that they don't want to assume you want the entire query result in memory so they make you store entries as a side effect. But I agree, very odd.